src/main/java/Duke.java

```diff
@@ -251,7 +108,7 @@ static void readFile(String filePath) throws FileNotFoundException {
                     DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy");
```

Comment:
Since the formatter is used multiple times, you can perhaps consider declaring it outside of the if-else statement. Also, since it is a constant, it might be good to name it in all caps.